### DISCUSSION TITLE
Deepin v23  常用软件包缺失

### DISCUSSION THREAD

**Discussion by iamhyc** [linuxdeepin/developer-center#4360] [General | 常规讨论]
目前在v23 beta版本中，仓库仍然存在Ubuntu/Debian下常用软件包缺失，目前个人遇到如下的缺失：

- [x] [ncdu](https://dev.yorhel.nl/ncdu), Ncdu is a disk usage analyzer with an ncurses interface. 
  > 实用的硬盘使用空间分析工具
- [ ] [iptsd](https://github.com/linux-surface/iptsd) (https://github.com/deepin-community/SIG/pull/135)
  > Userspace daemon for Intel Precise Touch & Stylus.
- [ ] [libwacom-surface](https://github.com/linux-surface/libwacom-surface) (https://github.com/deepin-community/SIG/pull/135)
  > Patches to support Microsoft Surface Devices with `libwacom`. 
- [ ] [v4l2loopback-dkms](https://github.com/umlaeute/v4l2loopback), a kernel module to create V4L2 loopback devices.
  > 用于创建虚拟摄像头。
- [ ] [libcamera](https://git.libcamera.org/libcamera/libcamera.git)
  > A complex camera support library for Linux, Android, and ChromeOS.

**Comment by BLumia** [comment]
我记得 ncdu @xzl01 打了来着（包源码仓 https://github.com/deepin-community/ncdu ）...

顺便，我们鼓励有兴趣维护软件包的社区小伙伴一起参与新软件包的打包与维护哦 :)

**Reply by iamhyc** [reply]
那么要如何参与呢😀看到那个`template-repository`仓库，但不知怎么下手

**Reply by BLumia** [reply]
可以参见 [deepin-community协作指南](https://wiki.deepin.org/deepin-community%E5%8D%8F%E4%BD%9C%E6%8C%87%E5%8D%97) 这个 wiki 页面的描述进行新包的递交以及参考后续的维护流程。另外强烈建议加入 sysdev SIG 的实时聊天频道（推荐使用 [Matrix](https://wiki.deepin.org/Matrix)），这样如果有问题不清楚的话可以更快进行交流与解决。当然继续在 GitHub discussion 提问也是可以的 :)

**Reply by iamhyc** [reply]
@BLumia 我看了一下这流程，然后参考了下 `deepin-community/ncdu` 这个仓库，就是把大象关冰箱里吗？

1. 申请创建仓库； 2. fork管理员创建的仓库，然后把带debian文件夹的源码copy进去；3. 提PR等deepin内部构建。

我有以下两个问题：
1. 原始仓库的git历史信息丢失：虽然这对构建没有影响，但是否给了维护者过大的权限？这并不安全。
2. 版本更新：如何保证与原始仓库代码的一致性？是否总是构建原始仓库的最新版本？这由谁来决定？有什么策略？

如果有需要的话，上面的问题我可以转发到 sysdev SIG。

**Reply by BLumia** [reply]
ncdu 已经有了（[在 unstable 仓库里](https://ci.deepin.com/repo/topics/beige/unstable/pool/main/n/ncdu/)），应该只需要提个集成。

> 就是把大象关冰箱里吗？
>
>> 申请创建仓库； 2. fork管理员创建的仓库，然后把带debian文件夹的源码copy进去；3. 提PR等deepin内部构建。

对。其中重点在 2，其中附带的需要是个规范的 debian 目录，不能只是打出包程度的。也就是说尤其是 `debian/rules`, `debian/copyright`, `debian/control`, `debian/changelog` 的内容需要规范的提供。至于构建，实际 PR 提交后，构建应该也是所有人都可以访问的（位于 <build.deepin.com> ）。

上面的三个步骤走完后，包就会出现在 unstable 仓库了，希望包出现在 testing/stable 的话需要提集成（参见之前那个 Wiki 的步骤四即可）。

> 我有以下两个问题：
>
> 1. 原始仓库的git历史信息丢失：虽然这对构建没有影响，但是否给了维护者过大的权限？这并不安全。

包维护仓库和软件项目本身的维护仓库性质上是不一样的。GitHub deepin-community 组织下的包维护仓库的性质与 salsa.debian.org 的性质是很近似的。版本控制跟进的是包维护的历史，而不是软件代码研发的历史。软件的研发仍然是在上游发生的。

> 2. 版本更新：如何保证与原始仓库代码的一致性？是否总是构建原始仓库的最新版本？这由谁来决定？有什么策略？

由维护者决策，如果影响比较大的话就需要共同决策了（组个 SIG 来维护，或者邮件列表讨论之类），影响非常大的话甚至需要发提案。不过就 ncdu 这类而言，应该基本没什么影响。可以考虑总是拉最新的。

另外，如果参与维护的话，还是强烈建议加入 sysdev 的 Matrix 群（或者对应的 Telegram 桥接群）的，这样如果存在拿不准的情况下可以更方便的发起讨论。

**Reply by iamhyc** [reply]
谢谢答复。

Matrix群还是类似内部讨论，我想这些流程规范相关的讨论还是放在Github上好。
虽然放这个discussion下也不是很合适

**Reply by xzl01** [reply]
> 谢谢答复。
> 
> Matrix群还是类似内部讨论，我想这些流程规范相关的讨论还是放在Github上好。 虽然放这个discussion下也不是很合适

Matrix群并不是内部讨论，我们选择matrix的原因是公开房间对所有人可见，也就是说你可以加入房间并且浏览全部的历史讨论，决策我们会在github公布并且寻求意见。

**Reply by BLumia** [reply]
算公开讨论，不过我们目前没有类似可公开检索的 irc log 的同类替代方案。

流程规范目前都主要放在 deepin wiki 了。在这里有相关的完整讨论供其他有兴趣的人检索也挺好的 :)

**Comment by xzl01** [comment]
@chenchongbiao 有兴趣把这几个用你的打包脚本跑跑？？

**Reply by iamhyc** [reply]
我先把 `iptsd` 和 `libwacom-surface` 两个包占位了🤣
